lib/rubocop_todo_corrector/bundle_installer.rb

```diff
@@ -51,10 +51,12 @@ def gemfile_content
       [
         "source 'https://rubygems.org'",
         *@gem_specifications.map do |gem_specification|
-          format(
-            "gem '%<gem_name>s', '%<gem_version>s'",
-            gem_specification
-          )
+          case gem_specification
+          in gem_version: nil
+            format("gem '%<gem_name>s'", gem_specification)
+          else
+            format("gem '%<gem_name>s', '%<gem_version>s'", gem_specification)
+          end
```

Comment:
Thank you for the pull request! It's not hard to imagine that testing this library was a little bit hard, so sorry about this.

It looks almost good, but I have not yet understood why this change is necessary in this timing. You added this change because it is necessary for this, right?
The version of each gem should always be written in the Gemfile.lock, and I can't imagine a situation where that is not available. Could you explain more about the background about of this change?

Reply:
Sure! At the point when I was trying to make the first commit the problem observed through testing was something like this:

```
/ruby/3.1.5/lib/ruby/3.1.0/rubygems/requirement.rb:106:in `parse':  (Bundler::Dsl::DSLError)
[!] There was an error parsing `Gemfile_rubocop_todo_corrector.rb`: Illformed requirement [""]. Bundler cannot continue.

 #  from /r7kamura/rubocop_todo_corrector/tmp/Gemfile_rubocop_todo_corrector.rb:5
 #  -------------------------------------------
 #  gem 'rubocop-rspec', '2.10.0'
 >  gem 'rubocop-rails-omakase', ''
 #  -------------------------------------------
```

The trouble was coming from the discrepancy between dummy_rubocop.yml and dummy_gemfile.lock, where the  dependency to rubocop-rails-omakase was detected in dummy_rubocop.yml, which was not present in dummy_gemfile.lock. This is where the assumption that the "version of each gem should always be written in the Gemfile.lock" turns out to not always be true, at least in the testing world.

It seemed to me there were two options to circumvent: 1. update dummy_gemfile.lock so it contains rubocop-rails-omakase, or 2. create a branch to check if gem_version is nil. My choosing the latter is because it seemed like you had expected that gem_version could be nil (so said the doc on GemVersionDetector#call), which I came across and "perhaps it is time to fix" was my response to it. I can see how I could have explained better, so thanks for pointing it out!

With that said, now I see that my second commit seems to have resolved the original error[^1] and my intention has become even less distinct 😣 Yes it's completely legitimate that you wondered why this portion should be here. But I'm on the fence if I should cancel it or leave it. Let me know if you have got any preference after this 🙂 

[^1]: Reverting the second commit and cancelling the said lines can reproduce though, for what it's worth:
    ```diff
     diff --git a/lib/rubocop_todo_corrector/bundle_installer.rb b/lib/rubocop_todo_corrector/bundle_installer.rb
     index e682cb3..98ad843 100644
     --- a/lib/rubocop_todo_corrector/bundle_installer.rb
     +++ b/lib/rubocop_todo_corrector/bundle_installer.rb
     @@ -51,12 +51,7 @@ module RubocopTodoCorrector
           [
             "source 'https://rubygems.org'",
             *@gem_specifications.map do |gem_specification|
    -          case gem_specification
    -          in gem_version: nil
    -            format("gem '%<gem_name>s'", gem_specification)
    -          else
    -            format("gem '%<gem_name>s', '%<gem_version>s'", gem_specification)
    -          end
    +          format("gem '%<gem_name>s', '%<gem_version>s'", gem_specification)
             end
           ].join("\n") << "\n"
         end
    diff --git a/spec/rubocop_todo_corrector/commands/describe_spec.rb b/spec/rubocop_todo_corrector/commands/describe_spec.rb
    index 7515625..462184d 100644
    --- a/spec/rubocop_todo_corrector/commands/describe_spec.rb
    +++ b/spec/rubocop_todo_corrector/commands/describe_spec.rb
    @@ -5,7 +5,7 @@ RSpec.describe RubocopTodoCorrector::Commands::Describe do
         subject do
           described_class.call(
             cop_name:,
    -        temporary_gemfile_path: 'Gemfile'
    +        temporary_gemfile_path: 'tmp/Gemfile_rubocop_todo_corrector.rb'
           )
         end
    ```

Reply:
I see, it seems natural for me to update spec/fixtures/dummy_gemfile.lock to include rubocop-rails-omakase, but this way of dealing with the issue seems to be a good idea for once